File: leetcode/205. Isomorphic Strings/isomorphic.js
const checkIsomorphic = function(s,t){
    var dict = new Object({})
    length = s.length
    for (let i = 0; i < length; i++) {
        if (s[i] in dict && dict[s[i]] !== t[i])
            return false
        else if (Object.values(dict).indexOf(t[i]) != -1 && !(s[i] in dict))
            return false
        else
            dict[s[i]] = t[i]
    }
    return true
}

const checkIsomorphic_2 = function(s,t){
    var sdict = new Object({})
    var tdict = new Object({})
    length = s.length
    for (let i = 0; i < length; i++) {
        if (!(s[i] in sdict))
            sdict[s[i]] = t[i]
        if (!(t[i] in tdict))
            tdict[t[i]] = s[i]
        if (sdict[s[i]] != t[i] || tdict[t[i]] != s[i])
            return false
    }
    return true
}

File: LeetCode/205. Isomorphic Strings/isomorphic.py
def isomorphic_strings(s, t):
    strings = dict()
    length = len(s)
    for i in range(length):
        if s[i] in strings and strings[s[i]] != t[i]:
            return False
        elif t[i] in strings.values() and s[i] not in strings:
            return False
        else:
            strings[s[i]] = t[i]
    return True;
    
s = "egeg"
t = "adzd"
print(isomorphic_strings(s, t))

def isomorphic_strings_1(s, t):
    sdict = dict()
    tdict = dict()
    length = len(s)
    for i in range(length):
        if s[i] not in sdict:
            sdict[s[i]] = t[i]
        if t[i] not in tdict:
            tdict[t[i]] = s[i]
        if s[i] in sdict and sdict[s[i]] != t[i] or t[i] in tdict and tdict[t[i]] != s[i]:
            return False
    return True
